api/tests/unit/application/session/index_test.js

```diff
@@ -498,6 +498,26 @@ describe('Unit | Application | Sessions | Routes', function () {
       // then
       expect(response.statusCode).to.equal(200);
     });
+
+    it('return forbiden access if user has METIER role', async function () {
+      // given
+      sinon.stub(sessionController, 'assignCertificationOfficer').returns('ok');
+
+      sinon.stub(securityPreHandlers, 'checkUserHasRoleMetier').resolves(true);
+      sinon.stub(securityPreHandlers, 'checkUserHasRoleSuperAdmin').resolves({ source: { errors: {} } });
+      sinon.stub(securityPreHandlers, 'checkUserHasRoleSupport').resolves({ source: { errors: {} } });
+      sinon.stub(securityPreHandlers, 'checkUserHasRoleCertif').resolves({ source: { errors: {} } });
+
+      const httpTestServer = new HttpTestServer();
+      await httpTestServer.register(moduleUnderTest);
+
+      // when
+      const response = await httpTestServer.request('PATCH', '/api/admin/sessions/1/certification-officer-assignment');
+
+      // then
+      expect(response.statusCode).to.equal(403);
+      sinon.assert.notCalled(sessionController.assignCertificationOfficer);
```

Comment:
j'ai l'impression qu'on teste le fonctionne des prehandler dans hapi en vérifiant que le controller n'est pas appelé, pas sure que ça apporte en plus et qu'on aie besoin de le stubber. Je teste en local

Reply:
On peut supprimer le stub sans pb, le teste passe. Je regarde la doc de hapi mais pour il me semble que le prehandler est joué avant l'appel au controller

Reply:
`The pre option allows defining methods for performing actions before the handler is called.` -> https://hapi.dev/api/?v=20.2.2#-routeoptionspre